src/components/follow-list/CheckBox.tsx

```diff
@@ -0,0 +1,45 @@
+import { ChangeEvent, useEffect } from 'react';
+
+interface CheckboxProps {
+  isAllChecked: boolean;
+  username: string;
+  checkList: string[];
+  setCheckList: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+export default function Checkbox({ isAllChecked, username, checkList, setCheckList }: CheckboxProps) {
+  let isChecked = false;
+
+  useEffect(() => {
+    isChecked = checkList.includes(username);
+  }, [isAllChecked]);
```

Comment:
useEffect 는 리액트 외부 세계를 다루기 위함이 아니라면, 사용하지 않아야 해요. (렌더링에 비효율적이고, 버그의 원천이 될 확률이 높아요)

공식 문서에서도 이 패턴을 지양하라는 내용을 찾아볼 수 있어요.
https://react.dev/learn/you-might-not-need-an-effect#caching-expensive-calculations

```suggestion
  const isChecked = useMemo(() => checkList.includes(username), [isAllChecked]);
```

Reply:
그리고 리액트 컴포넌트는 매 렌더링마다 새로 호출되기 때문에, 이렇게 변수를 선언하고 대입하면 isChecked 변수는 렌더링될 때 마다 다시 false로 초기화돼요. 컴포넌트 안에 리액트 렌더링 사이클에 구애받지 않는 변수를 선언하고 싶으면 useRef 를 활용해야 해요. (useRef 도 특별한 이유가 없는 한 사용하지 않아야 해요. 여기서의 특별한 이유는 애니메이션 구현, 리액트를 지원하지 않는 라이브러리의 직접 접근 등이 있어요.)